src/builtins/Transformers.jl

```diff
@@ -900,9 +900,7 @@ end
 
 # If v=categorical('a', 'a', 'b', 'a', 'c') and MMI.int(v[1]) = ref
 # then `_hot(v, ref) = [true, true, false, true, false]`
-_hot(v::AbstractVector{<:CategoricalValue}, ref) = map(v) do c
-    MMI.int(c) == ref
-end
+function hot(col, ref) map(col) do c  if typeof(c) != Missing MMI.int(c) == ref else missing end end end
```

Comment:
- for readability, please unroll the `do`-`end` block, as in the original code. The same applies to `if`-`else`-`end` blocks
- Turns out that the new `hot` function is not type-stable:

```julia
 col = [1, missing, 2, 1, 2] |> categorical
julia> @code_warntype hot(col, int(col[1]))
Variables
  #self#::Core.Const(hot)
  col::CategoricalArrays.CategoricalVector{Union{Missing, Int64}, UInt32, Int64, CategoricalArrays.CategoricalValue{Int64, UInt32}, Missing}
  ref::UInt32
  #9::var"#9#10"{UInt32}

Body::Any
1 ─ %1 = Main.:(var"#9#10")::Core.Const(var"#9#10")
│   %2 = Core.typeof(ref)::Core.Const(UInt32)
│   %3 = Core.apply_type(%1, %2)::Core.Const(var"#9#10"{UInt32})
│        (#9 = %new(%3, ref))
│   %5 = #9::var"#9#10"{UInt32}
│   %6 = Main.map(%5, col)::Any                  <--------------  Any type here
└──      return %6
```

(Incidentally, it's more readable if  instead of `typeof(c) != Missing` you do `ismissing(c)` - and both compile to the same code.) Maybe to fix the instability, you want to have two `hot` methods, one dispatched on `AbstractVector{<:CategoricalVector}` and one dispatched on `AbstractVector{<:Union{Missing, CategoricalValue}}`? In each case you know at compile time the output type.

Reply:
> for readability, please unroll the do-end block, as in the original code. The same applies to if-else-end blocks

I've done that in the latest commit.
 
> Turns out that the new hot function is not type-stable:

I'm not completely sure of what that is. But I suppose that it has to do with the map assuming `col` as type Any. But when I checked it on my REPL I didn't find Any as the type for col I'll check it again. But I guess not it won't happen that way as I've added the types for the hot function now (as you suggested). Let me know if it's ok.

Reply:
I think this is fine now, thanks.